File: back/src/main/java/com/clinic/entities/Stock.java
package com.clinic.entities;

import com.clinic.entities.keys.StockId;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import java.util.Calendar;

@Table(name = "stock")
@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Stock {
    @EmbeddedId
    private StockId stockId;

    private String name;

    private int amount;

    private String description;

    private int minAmount;

    @Temporal(TemporalType.TIMESTAMP)
    private Calendar lastUpdateTime;

    @JsonManagedReference
    @ManyToOne
    @JoinColumn(name = "last_update")
    private User user;
}

File: front/SAUT/src/app/_models/Stock.ts
import { Client, User } from "./Client";

export interface Stock{
    stockId: StockId;
    vendorId: number;
    name: string;
    amount: number;
    description: string;
    minAmount: number;
    lastUpdateTime: string;
    user: User
}

export interface StockId{
    id: number;
    vendor: number;
}